File: 2520.统计能整除数字的位数.js
/*
 * @lc app=leetcode.cn id=2520 lang=javascript
 *
 * [2520] 统计能整除数字的位数
 */

// @lc code=start
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function(num) {
    let t = num;
    let res = 0;
    while (t) {
        if (num % (t % 10) == 0) {
            res += 1;
        }
        t = Math.floor(t / 10);
    }
    return res;
};
// @lc code=end

File: 2520.统计能整除数字的位数.java
/*
 * @lc app=leetcode.cn id=2520 lang=java
 *
 * [2520] 统计能整除数字的位数
 */

// @lc code=start
class Solution {
    public int countDigits(int num) {
        int t = num;
        int res = 0;
        while (t != 0) {
            if (num % (t % 10) == 0) {
                res++;
            }
            t /= 10;
        }
        return res;
    }
}
// @lc code=end

